ivy/functional/ivy/layers.py

```diff
@@ -2340,3 +2340,172 @@ def _get_num_padded_values(i, p, n, k, s):
     return max(0, left_padding - current_index) + max(
         0, current_index + k - n - left_padding
     )
+
+
+def _bilinear_interpolate(
```

Comment:
would `ivy.interpolate` with the `bilinear` mode be useful here? Just wanted to avoid code duplication if that's the case.
If not, we should rename this helper a bit to avoid confusion with the existing `interpolate` implementation

Reply:
IIRC one main difference was `sampling_ratio`, being available in [`roi_align`](https://pytorch.org/vision/main/generated/torchvision.ops.roi_align.html#torchvision.ops.roi_align), which allows adaptive number of samples based on size of `roi`. Does `_roi_bilinear_interpolate` sounds more clear?

Reply:
oh okay, ig we could extend the existing `bilinear` mode in the `interpolate` function if that's the case. @sherry30 any thoughts? Thanks 😄

Reply:
sounds good, let's add a ToDo here to integrate this logic into `ivy.interpolate` then @ShreyanshBardia, thanks @sherry30 😄